### PR TITLE
VFB-10 Add driver overview PDF

### PR DESCRIPTION
**Pull request by DanWilkins2107**
Implemented a driver overview PDF component. This will take in data, a driver name and a date, and will return a pdf component that can be viewed with PDF viewer or downloaded.

- [x] Make sure you've verified it works via `npm run dev`
- [x] Make sure you've linted via `npm run lint`
    - can run `npm run lint_fix` to try and fix as any mistakes automatically as possible!
- [x] Make sure you've tested via `npm run test`

### CODE REVIEW

File: src/components/DriverOverview/DriverOverview.tsx [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+        <Document>
+            <Page size="A4" style={[styles.container, styles.flexColumn]}>
+                <View style={[styles.infoAndLogoContainer, styles.flexRow]}>
+                    <View style={styles.informationContainer}>
+                        <Text style={styles.h1text}>Driver Overview</Text>
+                        <Text style={styles.h2text}>Driver Name: {driverName}</Text>
+                        <Text style={styles.h2text}>Date: {dateConverter(date)} </Text>
+                    </View>
+                    {/* es lint disable needed to remove the need for alt text on the logo */}
+                    {/* eslint-disable-next-line */}
```
Comment (SamYap-Softwire) [inline src/components/DriverOverview/DriverOverview.tsx:173]: can you specify the eslint rule that is being disabled here?

**Review by SamYap-Softwire — CHANGES_REQUESTED** [review]
Seems good :) Just leaving it as request changes in case you merge the temporary page as well

File: src/components/DriverOverview/DriverOverview.tsx [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    },
+});
+
+const DriverOverviewCard: React.FC<DriverOverviewCardProps> = ({ driverName, date, data }) => {
+    const dateConverter: (date: Date) => string = (date) => {
+        const day = date.getDate();
+        const month = date.getMonth() + 1;
+        const year = date.getFullYear();
+        return `${day}/${month}/${year}`;
+    };
```
Comment (daniel-conlan-softwire) [inline src/components/DriverOverview/DriverOverview.tsx:110]: Can the signature of this function not be simplified to `const dataConverter = (date: Date): string => {...}`?

Also, I take it that the desired behaviour of this function is not to have the month and day padded by zeroes?

**Review by daniel-conlan-softwire — COMMENTED** [review]
Looks all good to me - just had a small note on your `dateConverter` function

**Review by daniel-conlan-softwire — APPROVED** [review]
Looks all good to me - just had a small note on your dateConverter function. (I had meant for the previous review to be an approval)

**Review by Var1377 — APPROVED** [review]

File: src/components/DriverOverview/DriverOverview.tsx [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+    flexColumn: {
+        flexDirection: "column",
+        display: "flex",
+    },
+    flexRow: {
+        flexDirection: "row",
+        display: "flex",
+        justifyContent: "space-between",
+    },
+    tableHeader: {
```
Comment (SamYap-Softwire) [inline src/components/DriverOverview/DriverOverview.tsx:95]: You can probably use `flexRow` & `tableHeader` for this (unless you don't want `justifyContent: space-between`?

**Review by SamYap-Softwire — DISMISSED** [review]
looks good - just one minor change

File: src/components/DriverOverview/DriverOverview.tsx (hunk 2 above)
Comment (DanWilkins2107) [reply src/components/DriverOverview/DriverOverview.tsx:128]: Yep I don't really want the padding on it! and you are correct!

File: src/components/DriverOverview/DriverOverview.tsx [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+    parcelsColumnWidth: {
+        width: "10%",
+    },
+    instructionsColumnWidth: {
+        width: "25%",
+    },
+});
+
+const DriverOverviewCard: React.FC<DriverOverviewCardProps> = ({ driverName, date, data }) => {
+    const dateConverter = (date: Date): string => {
```
Comment (paopaosoftwire) [inline src/components/DriverOverview/DriverOverview.tsx:121]: I think you can do something similar with toLocaleDateString() which should be more concise?

**Review by paopaosoftwire — CHANGES_REQUESTED** [review]

File: src/components/DriverOverview/DriverOverview.tsx (hunk 4 above)
Comment (DanWilkins2107) [reply src/components/DriverOverview/DriverOverview.tsx:121]: You can yep!

**Review by paopaosoftwire — APPROVED** [review]
Good stuff. Talked about the repeated classes in person and agreed that its easier to read and edit so we'll leave it as is.

File: public/logo.png
Comment (TomCusack-Softwire) [inline public/logo.png:1]: same comment on Sam's ticket -
https://github.com/VauxhallFoodbank/vauxhall-foodbank/pull/54/files#r1297031920

File: src/components/DriverOverview/DriverOverview.tsx [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+    instructions?: string;
+}
+
+const styles = StyleSheet.create({
+    container: {
+        padding: 25,
+        alignItems: "center",
+    },
+    infoAndLogoContainer: {
+        height: "100",
```
Comment (TomCusack-Softwire) [inline src/components/DriverOverview/DriverOverview.tsx:33]: %? units? can this just be a number if react-pdf does it for you?

File: src/components/DriverOverview/DriverOverview.tsx [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+        alignSelf: "flex-start",
+        margin: 10,
+        padding: 10,
+    },
+    h1text: {
+        fontSize: 24,
+        paddingLeft: 10,
+        paddingRight: 20,
+        paddingBottom: 5,
+        fontFamily: "Helvetica",
```
Comment (TomCusack-Softwire) [inline src/components/DriverOverview/DriverOverview.tsx:47]: is it possible to set fontFamily in 'the root' once?

File: src/components/DriverOverview/DriverOverview.tsx [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+    },
+    h2text: {
+        fontSize: 12,
+        paddingLeft: 10,
+        fontFamily: "Helvetica",
+    },
+    logoStyling: {
+        maxHeight: 60,
+        maxWidth: 60 * 1.7,
```
Comment (TomCusack-Softwire) [inline src/components/DriverOverview/DriverOverview.tsx:56]: where does 1.7 come from?

File: src/components/DriverOverview/DriverOverview.tsx [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+    );
+
+    const table = data.map((rowData, index) => {
+        return (
+            <View key={index} style={[styles.tableRow, styles.flexRow]} wrap={false}>
+                <View style={[styles.tableColumn, styles.nameColumnWidth]}>
+                    <Text>{rowData.name}</Text>
+                </View>
+                <View style={[styles.tableColumn, styles.addressColumnWidth]}>
+                    <Text>{rowData.address.line1}</Text>
```
Comment (TomCusack-Softwire) [inline src/components/DriverOverview/DriverOverview.tsx:151]: what happens if eg line2 is blank? does `<Text>` collapse to nothing or does it leave a blank line?

File: src/components/DriverOverview/DriverOverview.tsx [hunk 9]
```diff
@@ -0,0 +1,208 @@
...
+                        <Text style={styles.h1text}>Driver Overview</Text>
+                        <Text style={styles.h2text}>Driver Name: {driverName}</Text>
+                        <Text style={styles.h2text}>Date: {date.toLocaleDateString()} </Text>
+                    </View>
+                    {/* eslint-disable-next-line -- needed to remove the need for alt text on the logo */}
+                    <Image src="/logo.png" style={styles.logoStyling}></Image>
+                </View>
+                <View style={styles.warningSection}>
+                    <Text style={styles.h3text}>
+                        At the end of your shift, please call/text the Dispatch phone to let us know
```
Comment (TomCusack-Softwire) [inline src/components/DriverOverview/DriverOverview.tsx:187]: maybe this text should be in new table... (this will be another ticket!)

**Review by TomCusack-Softwire — CHANGES_REQUESTED** [review]

File: src/components/DriverOverview/DriverOverview.tsx (hunk 5 above)
Comment (DanWilkins2107) [reply src/components/DriverOverview/DriverOverview.tsx:33]: Yes this is pt, the default setting for react-pdf

File: src/components/DriverOverview/DriverOverview.tsx (hunk 7 above)
Comment (DanWilkins2107) [reply src/components/DriverOverview/DriverOverview.tsx:56]: The aspect ratio for the logo. Will add a comment saying this

File: src/components/DriverOverview/DriverOverview.tsx (hunk 9 above)
Comment (DanWilkins2107) [reply src/components/DriverOverview/DriverOverview.tsx:187]: Probably should be!

File: src/components/DriverOverview/DriverOverview.tsx (hunk 8 above)
Comment (DanWilkins2107) [reply src/components/DriverOverview/DriverOverview.tsx:151]: The line collapses to nothing.

File: src/components/DriverOverview/DriverOverview.tsx (hunk 7 above)
Comment (TomCusack-Softwire) [reply src/components/DriverOverview/DriverOverview.tsx:56]: I think I would rather this just be
`maxWidth: 102, // maintains aspect ratio of logo`

File: src/components/DriverOverview/DriverOverview.tsx (hunk 5 above)
Comment (TomCusack-Softwire) [reply src/components/DriverOverview/DriverOverview.tsx:34]: should this just be a number (not a string)?

**Review by TomCusack-Softwire — APPROVED** [review]
see changes in comments above

**Pull request merged**